axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java

```diff
@@ -479,10 +479,10 @@ public void onCompleted() {
 
         @Override
         public void onError(Throwable reason) {
-            logger.warn("Error on connection from client: {}", reason.getMessage());
-            if (eventStoreRequestObserver != null) {
-                StreamObserverUtils.complete(eventStoreRequestObserver);
+            if (!GrpcExceptionBuilder.isCancelled(reason)) {
+                logger.warn("Error on connection from client: {}", reason.getMessage());
             }
+            StreamObserverUtils.complete(eventStoreRequestObserver);
```

Comment:
Looking at the code seems that `StreamObserverUtils.complete(eventStoreRequestObserver);` always need `removeTrackerInfo();` called next.
If this is true, I will suggest to extract this two call as a single private method.